src/Controls/src/Core/Button.cs

```diff
@@ -466,6 +458,12 @@ void ITextElement.OnTextTransformChanged(TextTransform oldValue, TextTransform n
 		public virtual string UpdateFormsText(string source, TextTransform textTransform)
 			=> TextTransformUtilites.GetTransformedText(source, textTransform);
 
+		void ICommandElement.CanExecuteChanged(object sender, EventArgs e) =>
+			RefreshIsEnabledProperty();
+
+		internal override bool IsEnabledCore =>
```

Comment:
semi-related questions on this one

- Thoughts on just checking if the type implements `ICommandElement` inside VisualElement and then running the logic that's duplicated across `Button/ImageButton/SearchBar`? 
- What's the 3rd party story here? What if I want to create 

```C#
ShanesButton : VisualElement
```
And then opt into this same coercion. 

- If we leave it this way.  I think this could be changed to `private protected virtual bool IsEnabledCore` instead of `Internal`?

Reply:
it could just be plain protected but that is a new public API... Maybe for net8 we just expose it?

Reply:
If this is the API we want to make public to enable this path, then it seems like it makes sense to make it protected here and then on the back port we make it private protected.  

Unless there's something more magical, we are hoping to add in net8 to make this a bit more streamlined.

Reply:
Along these lines, one thing I'm curious about is if it makes sense to think about this in terms of structures we've used elsewhere.  Like, I noticed in VE you've taken the concepts we have inside the `PropertyPropagationExtensions` and moved it out of there. 

![image](https://user-images.githubusercontent.com/5375137/212206913-0f18dc64-6841-42c7-b53f-9ed114da3cd3.png)

And then this code from `MenuItem`
![image](https://user-images.githubusercontent.com/5375137/212207045-16597bd2-c04e-4894-91e6-8f69f03f627f.png)

is basically doing the same thing that the code inside Image/Button/SearchBar when they call up to `RefreshIsEnabledProperty` 

We have a few places that have an `IsEnabledProperty` but they inherit from `Element`

`Cell` and `BaseShellItem` are the main two that I'm looking at currently. 

If we keep with tradition, then I think we'd be adding something like 

```C# 
interface IEnablingController
{
     bool UserSetValue
}
```

Then inside `PropertyPropagationExtensions` we'd have something similar to all the other propagate blocks in there


```C#
		internal static void PropagateIsEnabled(Element target, Element source)
		{
			var controller = target as IEnablingController;
			if (controller == null)
				return;

			var sourceController = source as IEnablingController;
			CommandElement.RefreshPropertyValue(sourceController, IsEnabledProperty, _isEnabledExplicit);
		}
```

Not saying the `PropagationExtentions` are a great pattern but it's the pattern we've been using. So, it feels weird to deviate unless we're deviating all of it to a better home. 

Or we just merge this as is and create a .NET8 discussion item :-)

Reply:
Yeah, I originally had this as the propagation, but then @hartez pointed out it is not the same because we never copy anything from the parent/source. It is a re-evaluation. And then also, it would propagate again because IsEnabledChanged triggers propagation.

Reply:
>  I realize that's most likely because the scoping of _isEnabledExplicit.

Mostly that, and partly that the whole pattern of externalizing propagation into those extensions seems extraneous. Pretty sure that in every instance, the VisualElement has all the info it needs to update the value (either from itself or via its Parent). But specifically for _this_ instance, there's no way to handle the propagation without access to `_isEnabledExplicit.` 

I agree that `RefreshPropertyValue` shouldn't be on CommandElement.

Reply:
> Pretty sure that in every instance, the VisualElement has all the info it needs to update the value

It seems like the rules of `IsEnabled` matter beyond `VisualElement`. In this PR we are copying part of the logic over to `MenuItem` which inherits from `Element`. I think this logic is also relevant to `Cell` so that `IsEnabled` can propagate through `Cells` into `Elements` as well as `BaseShellItem` propagating down. 

> partly that the whole pattern of externalizing propagation into those extensions seems extraneous.

I could see that. Though, part of my motivation is to contain the logic that defines "IsEnabled" propagation and calculations. `VisualElement` (partial classes included) is massive. I don't think externalizing certain workflows is the worst idea. Having the external `EnablingElement` organizes our rule sets into a single place. As is, we have logic duplicated across every control that implements "ICommandElement" and then same for the ones that use "IEnabledProperty" 

Also, it feels like it matches the patterns we've used other places where we have common BP's that we need to sprinkle around places.

Reply:
Maybe at this point we merge this PR to fix the bugs, and then work on a better PR later to sort out these details - along with the Cell and things?

Reply:
@mattleibow sounds good. 

Can we make IsEnabledCore `protected virtual` and then on the back port make it private protected? 

Since we're following WPFs model [here](https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/ContentElement.cs#L636-L646) it seems like that makes sense.

Reply:
Done and done!